libraries/AP_Logger/AP_Logger_File.cpp

```diff
@@ -1057,7 +1057,7 @@ bool AP_Logger_File::io_thread_alive() const
     // disk.  Unfortunately these hardware devices do not obey our
     // SITL speedup options, so we allow for it here.
     SITL::SIM *sitl = AP::sitl();
-    if (sitl != nullptr) {
+    if (sitl != nullptr && sitl->speedup > 0) {
```

Comment:
Shouldn't we be using `get_speedup()` here?

Tim pointed out that he uses speedup on the command-line; if he does that then the number used here would be incorrect?

Reply:
Yes, using get_speedup() would be better but I didn't see any easy way for AP_Logger to reach down into a SIM::AIRCRAFT class.